src/Transition/CreateNewIDFUsingRulesV9_4_0.f90

```diff
@@ -1221,6 +1222,26 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                     EXIT
                   ENDIF
                 ENDDO
+                ! Begin - Special section for v9.4
+                IF (CurArgs .GE. 3) THEN
+                  changeMeterNameFlag = .true.
+                  DO numMeterCustom=1, totMeterCustom + totMeterCustomDecr
+                    MeterCustomName = MeterCustomNames(numMeterCustom)
+                    IF (MeterCustomName .eq. MakeUPPERCase(InArgs(3))) THEN
+                      changeMeterNameFlag = .false.
+                    END IF
+                  END DO
+                  IF (changeMeterNameFlag) THEN
+                    CALL ReplaceFuelNameWithEndUseSubcategory(OutArgs(3), NoDiff)
+                  END IF
+                END IF
```

Comment:
The special section for 9.4 is good up to here - this catches any changes needed for field A3 Source Meter Name (and avoids breaking the name if it came from a user-defined custom meter).

Reply:
Thank you for the review. I'll update the code and push it again.

Reply:
Sorry to chip in. But shouldn't the 4s be 3s here? The 3rd field is `Source Meter Name` and its value should also be updated according to Report Variables 9-3-0 to 9-4-0.csv?

```
                CurVar=4   ! In case Source Meter would change
                DO Var=4,CurArgs,2
```

Reply:
@hongyuanjia No. Field 3 is handled in it's own block, lines 1224:1235.

Reply:
@mjwitte, thanks for the clarification. So only the source meter name will be changed if necessary. Previously I was thinking this code block should replace fuel names for all possible meters, including those listed  `Output Variable or Meter Name X`.

Reply:
@hongyuanjia You may be right. I suppose a custom meter could reference other custom meters or a user-defined end-use meter. @dareumnam will build test files for more use-cases.